app/models/beer.rb

```diff
@@ -1,59 +1,45 @@
+# beer model
 class Beer < ApplicationRecord
   validates_presence_of :name, :beer_type, :brand
   validates_uniqueness_of :name
   has_many :user_beers
   has_many :users, through: :user_beers
   has_many :ratings
 
-  class << self
-    def search(params)
-      beers = scope_beers(params)
-      beers = beers.where(beer_type: params["type"]) unless ["all types", ""].include?(params["type"])
-      beers = check_text(params["text"], beers) if params["text"].present?
-      beers = sort_by(beers, params["sort"]) if params["sort"].present?
-      beers = calculate_offset(beers, params) if params[:page].present?
-      beers = beers.order(updated_at: :desc).limit(24)
-      beers
-    end
-
-    def check_text(text, beers)
-      beers.where("name LIKE ?", "%#{text.downcase}%")
-    end
-
-    def scope_to_user(token)
+  scope :current_beers, (lambda do |beer, token = nil|
+    if beer == 'my beers'
       User.find_by(password_digest: token).beers
```

Comment:
This hard dependency on `User` isn't too good. It's better to access it through the relation via `beers.users` or to scope it in the controller via Beers like: `Beers.joins(:users).where(:users => {id: 123})`. It's a shortcut to make a param that passes in `my_beers`, but your user will have an `id: 123` which is better to use.